### PR TITLE
Add rcf score in ThresholdingResult

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are a lot of dependencies, I only list the main class and test code to save reviewers' time. The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big one in the end and merge once after all review PRs get approved.

*Issue #, if available:*

*Description of changes:*
This PR adds an rcf score in ThresholdingResult so that we can use it when saving anomaly results.  Previously we don't need it since AnomalyResultTransportAction runs rcf first
and then threshold model. But for multi-entity result transport action, it sends the da
ta and gets back the ThresholdingResult, nothing in between.

Testing done:
1. Its callers have covered its usage in their unit tests.
2. Manual testing passes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by yizheliu-amazon — APPROVED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ThresholdingResult.java [hunk 1]
```diff
@@ -24,16 +24,21 @@
...
      * Constructor with all arguments.
      *
      * @param grade anomaly grade
      * @param confidence confidence for the grade
+     * @param rcfScore rcf score associated with the grade and confidence. Used
+     *   by multi-entity detector to differentiate whether the result is worth
+     *   saving or not.
      */
-    public ThresholdingResult(double grade, double confidence) {
+    public ThresholdingResult(double grade, double confidence, double rcfScore) {
```
Comment (ohltyler) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ThresholdingResult.java:38]: if this is only used for multi-entity detectors, shouldn't we still support the existing constructor, and have an additional constructor that takes in the `rcfScore` param?

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ThresholdingResult.java:38]: This is not only used for multi-entity detectors.  Single-entity detectors use it as well.

Comment (ohltyler) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ThresholdingResult.java:38]: Sorry, I mean, if adding the `rcfScore` is only used for multi-entity detectors, shouldn't we keep the old constructor and add this new constructor that takes in the additional `rcfScore` param?

Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ThresholdingResult.java:38]: Currently single-entity detectors get support from the constructor.  I don't break things.

Comment (ohltyler) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ThresholdingResult.java:38]: Got it, wanted to confirm

**Review by ohltyler — APPROVED** [review]

**Pull request closed without merging**